crates/pixi_manifest/src/task.rs

```diff
@@ -44,6 +56,55 @@ impl From<TaskName> for String {
     }
 }
 
+/// A task dependency with optional args
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize)]
+pub struct Dependency(pub TaskName, pub Option<Vec<String>>);
+
+impl Dependency {
+    pub fn name(&self) -> &TaskName {
+        &self.0
+    }
+
+    pub fn args(&self) -> &Option<Vec<String>> {
+        &self.1
+    }
+}
+
+impl From<TaskName> for Dependency {
+    fn from(name: TaskName) -> Self {
+        Dependency(name, None)
+    }
+}
+
+impl From<Dependency> for (TaskName, Option<Vec<String>>) {
+    fn from(dep: Dependency) -> Self {
+        (dep.0, dep.1)
+    }
+}
```

Comment:
I think creating a method `new` which takes two arguments makes more sense than this.